src/flamenco/types/fd_types.c

```diff
@@ -1521,8 +1520,9 @@ int fd_stake_history_decode_preflight(fd_bincode_decode_ctx_t * ctx) {
 void fd_stake_history_decode_unsafe(fd_stake_history_t* self, fd_bincode_decode_ctx_t * ctx) {
   ulong fd_stake_history_treap_len;
   fd_bincode_uint64_decode_unsafe(&fd_stake_history_treap_len, ctx);
-  self->pool = fd_stake_history_pool_alloc( ctx->valloc );
-  self->treap = fd_stake_history_treap_alloc( ctx->valloc );
+  ulong fd_stake_history_treap_max = fd_ulong_max( fd_stake_history_treap_len, FD_STAKE_HISTORY_MIN );
+  self->pool = fd_stake_history_pool_alloc( ctx->valloc, fd_stake_history_treap_max );
```

Comment:
> A malicious user could provide a oversize structure in instruction data though. Similarly, a fuzzer could provide a corrupted vote state, though this is assumed to not be possible on a real network.
>
> We did not handle this case correctly and would instead fail to deserialize the input (potentially leading to a different execution result).

Just so I understand, is the "potentially different execution result" you're referring to here a case where Agave succeeds and we fail? I imagine it's possible to construct a case where Agave "drops" the extra values, and we fail at the deserialization stage. Just curious as to exactly what you saw whilst fuzzing.

Reply:
Thanks for the context!